BenadrylAutomation/src/main/java/com/jnj/pageobjects/WhereToBuyPage.java

```diff
@@ -0,0 +1,203 @@
+
+	@FindBy(css = "#__ps-map-location-textbox_0")
+	private WebElement locationInput;
+
+	@FindBy(xpath = "//div[@class='ps-address']/div[1]/div[1]")
+	private List<WebElement> addresses;
+
+	private WebElement getOption(String option) {
+		return driver.findElement(By.xpath("//li[@role='option']/span[contains(text(),'" + option + "')]"));
+	}
+
+	private WebElement getStoreViewbtn(String storeName) {
+		return driver.findElement(By.xpath("//button[contains(@aria-label,'" + storeName + "')]"));
+	}
+
+	/**
+	 * Function to verifyheading on the Where to buy page
+	 * 
+	 * @param expectedHeader pass expected header
+	 */
+	public void verifyHeading(String expectedHeader) {
+		actualText = headingText.getText().trim();
+		expectedHeader = expectedHeader.trim();
+		Action.printAndAssert(actualText, expectedHeader);
+		BaseClass.extentInfoLog("Succesfully verifed heading : " + actualText);
+	}
+
+	/**
+	 * Function to initiate product search
+	 * @param productName pass product name
+	 */
+	public void initiateProductSearch(String productName) {
+		Action.explicitWait(specificProductSearchBox, 10);
+		Action.performActionwithExtentInfoLog(headingText, "sendKeys", "Sending keys to product search", productName);
+	}
+
+	/**
+	 * Function to search a product
+	 * 
+	 * @param productKey pass product name to search
+	 */
+	public void searchProduct(String productKey) {
+		Action.explicitWait(specificProductSearchBox, 20);
+		Action.explicitWaitForElementToDisappear(spinner, 20);
+		Action.performActionwithExtentInfoLog(specificProductSearchBox, "click", "Clicked inside product searchbox");
+		Action.waitFor(2000);
+		Action.performActionwithExtentInfoLog(specificProductSearchBox, "sendKeys",
+				"Searching for products " + productKey, productKey);
+		Action.explicitWait(prodResultListbox, 10);
+		Action.performActionwithExtentInfoLog(this.getOption(productKey), "click",
+				"Selecting product from listbox : " + productKey);
+		Action.explicitWait(resProductName, 10);
+		Action.printAndAssert(resProductName.getText().trim(), productKey);
+		BaseClass.extentInfoLog("Verifed product name after search to be, '", productKey + "'");
+	}
+
+	/**
+	 * Function to veriy 'Category' and 'Size' dropdown are displayed
+	 */
+	public void verifyCategoryAndSize() {
+		Assert.assertTrue(categoryDropdown.isDisplayed(), "Category dropdown not displayed");
+		BaseClass.extentInfoLog("Category dropdown displayed ");
+		Assert.assertTrue(sizeDropdown.isDisplayed(), "Category dropdown not displayed");
+		BaseClass.extentInfoLog("Size dropdown displayed ");
+	}
+
+	/**
+	 * Function to set new category and size value
+	 * 
+	 * @param category pass category text
+	 * @param size     pass size text
+	 */
+	public void changeProductOptions(String category, String size) {
+		if (category != null) {
+			Action.explicitWait(categoryDropdown, 10);
+			Action.selectByVisibleText(category, categoryDropdown, "Selecting category as " + category);
+		}
+		Action.waitFor(2000);
```

Comment:
@goutamnaik007 this is needed?

Reply:
yes. there are two dropdowns here. Once the value in first dropdown is selected, the second dropdown updates which takes approximately 1-2 seconds
@vaibhavNagvekarSJI

Reply:
fixed